pypeit/core/skysub.py

```diff
@@ -735,7 +735,9 @@ def local_skysub_extract(sciimg, sciivar, tilts, waveimg, global_sky, rn2_img,
                                             last_profile, box_rad, sobjs[iobj])
                     # If the extraction is bad do not update
                     if sobjs[iobj].OPT_MASK is not None:
-                        if sobjs[iobj].OPT_MASK.any():
+                        # if sobjs[iobj].OPT_MASK.any():
+                        # if there is only one good pixel `extract.fit_profile` fails
+                        if np.where(sobjs[iobj].OPT_MASK == True)[0].size > 5:
```

Comment:
Please make this:

if np.sum(sobjs[iobj].OPT_MASK) > extract_good_frac*nspec: 
   etc. 

This removes the arbitrariness of the number 5. Please add extract_good_frac as  keyword parameter in local_skysub_extract, document it, and set the default to what you think an appropriate number is, which is I guess is something like 0.01.

Reply:
Thanks for the feedback @jhennawi 
I made the changes that you suggested!